nginx_entrypoint.bash

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+set -euo pipefail
+
+mongodb_frontend_port=`printenv MONGODB_FRONTEND_PORT`
+mongodb_backend_host=`printenv MONGODB_BACKEND_HOST`
+mongodb_backend_port=`printenv MONGODB_BACKEND_PORT`
+bdb_frontend_port=`printenv BIGCHAINDB_FRONTEND_PORT`
+bdb_backend_host=`printenv BIGCHAINDB_BACKEND_HOST`
+bdb_backend_port=`printenv BIGCHAINDB_BACKEND_PORT`
+mongodb_whitelist=`printenv MONGODB_WHITELIST`
+dns_server=`printenv DNS_SERVER`
+
+nginx_health_check_port=`printenv NGINX_HEALTH_CHECK_PORT`
+
+threescale_secret_token=`printenv THREESCALE_SECRET_TOKEN`
+threescale_service_id=`printenv THREESCALE_SERVICE_ID`
+threescale_version_header=`printenv THREESCALE_VERSION_HEADER`
+threescale_provider_key=`printenv THREESCALE_PROVIDER_KEY`
+threescale_frontend_api_dns_name=`printenv THREESCALE_FRONTEND_API_DNS_NAME`
+threescale_upstream_api_port=`printenv THREESCALE_UPSTREAM_API_PORT`
+
+# sanity checks TODO(Krish): hardening
+if [[ -z "${mongodb_frontend_port}" || \
+    -z "${mongodb_backend_host}" || \
+    -z "${mongodb_backend_port}" || \
+    -z "${bdb_frontend_port}" || \
+    -z "${bdb_backend_host}" || \
+    -z "${bdb_backend_port}" || \
+    -z "${dns_server}" || \
+    -z "${nginx_health_check_port}" || \
+    -z "${threescale_secret_token}" || \
+    -z "${threescale_service_id}" || \
+    -z "${threescale_version_header}" || \
+    -z "${threescale_provider_key}" || \
+    -z "${threescale_frontend_api_dns_name}" || \
+    -z "${threescale_upstream_api_port}" ]] ; then
+  echo "Invalid environment settings detected. Exiting!"
+  exit 1
+fi
+
+NGINX_LUA_FILE=/usr/local/openresty/nginx/conf/nginx.lua
+NGINX_CONF_FILE=/usr/local/openresty/nginx/conf/nginx.conf
+
+# configure the nginx.lua file with env variables
+sed -i "s|SERVICE_ID|${threescale_service_id}|g" $NGINX_LUA_FILE
+sed -i "s|THREESCALE_RESPONSE_SECRET_TOKEN|${threescale_secret_token}|g" $NGINX_LUA_FILE
```

Comment:
A similar comment could be made about all the sed lines: maybe just look them all over to confirm that none will include the `|` (pipe) character.